skill-tree/src/main/java/com/RDS/skilltree/Endorsement/EndorsementServiceImpl.java

```diff
@@ -33,6 +51,55 @@ public Page<EndorsementModel> getEndorsements(PageRequest pageRequest) {
         return endorsementRepository.findAll(pageRequest);
     }
 
+    public Page<EndorsementModelFromJSON> getEndorsementsFromDummyData(PageRequest pageRequest, String skillIDString, String userIDString) throws IOException {
+        try {
+            List<EndorsementModelFromJSON> endorsementModelFromJSONList = readEndorsementsFromJSON();
+
+            if (endorsementModelFromJSONList.isEmpty()) {
+                return Page.empty(pageRequest);
+            }
+
+            List<EndorsementModelFromJSON> filteredEndorsements = filterEndorsements(endorsementModelFromJSONList, skillIDString, userIDString);
+            return createPagedEndorsements(filteredEndorsements, pageRequest);
+        } catch (IOException e) {
+            logger.error("Error reading endorsements JSON data: {}", e.getMessage());
+            throw new IOException("Error reading endorsements JSON data", e);
+        }
+    }
+
+    private List<EndorsementModelFromJSON> readEndorsementsFromJSON() throws IOException {
+        ClassPathResource resource = new ClassPathResource(dummyEndorsementDataPath);
+        return objectMapper.readValue(resource.getInputStream(), new TypeReference<List<EndorsementModelFromJSON>>() {});
+    }
+
+    private List<EndorsementModelFromJSON> filterEndorsements(List<EndorsementModelFromJSON> endorsements, String skillIDString, String userIDString) {
+        return endorsements.stream()
+                .filter(endorsement -> isMatchingSkillId(endorsement, skillIDString))
+                .filter(endorsement -> isMatchingUserId(endorsement, userIDString))
+                .collect(Collectors.toList());
+    }
+
+    private boolean isMatchingSkillId(EndorsementModelFromJSON endorsement, String skillIDString) {
+        return skillIDString == null || skillIDString.isEmpty() || endorsement.getSkillId().equals(UUID.fromString(skillIDString));
+    }
+
+    private boolean isMatchingUserId(EndorsementModelFromJSON endorsement, String userIDString) {
+        return userIDString == null || userIDString.isEmpty() || endorsement.getUserID().equals(UUID.fromString(userIDString));
+    }
+
+    private Page<EndorsementModelFromJSON> createPagedEndorsements(List<EndorsementModelFromJSON> endorsements, PageRequest pageRequest) {
+        int startIdx = (int) pageRequest.getOffset();
```

Comment:
`       @RequestParam(name = "offset", defaultValue = "0", required = false) @Min(0) int offset,
        @RequestParam(name = "limit", defaultValue = "10", required = false) @Min(1) int limit,
`

offset and limit have these default values (0 and 10) , will this prevent them from being null? tested a few requests with not passing them and could not get them to be  null. Any thoughts / scenarios where this might happen?
Also, since pagination was already existing , did not really modify these default values

Reply:
Marking this as resolved. Let me know if any concerns , we can re open it